### PR TITLE
Add react switch to code-gen

### LINKED ISSUE #720: Add React codegen configuration switch
Add here: https://github.com/neo-one-suite/neo-one/blob/master/packages/neo-one-server-plugin-project/src/types.ts#L33. Disable by default. Use the flag here: https://github.com/neo-one-suite/neo-one/blob/master/packages/neo-one-server-plugin-project/src/build/generateCommonCode.ts

Find all the places where we expect react code to be generated, e.g. the courses, and make sure we set that flag to true.

### PR DESCRIPTION
**Pull request by davemneo**
### Requirements

Add ability to switch from/between react and other ui framework options (or none)

### Description of the Change

basic setup for adding additional code generators. 

### Test Plan

none at this time. 

### Benefits

The ability to output a specific framework

### Applicable Issues

resolves #720

### CODE REVIEW

File: packages/neo-one-server-plugin-project/src/types.ts [hunk 1]
```diff
@@ -30,13 +30,26 @@ export interface NEOTrackerRequestOptions {
 
 export type RequestOptions = NetworkRequestOptions | SourceMapsRequestOptions | NEOTrackerRequestOptions;
 
+export enum CodegenLanguage {
+  JavaScript,
+  TypeScript,
+}
+
+export enum CodegenFrameworks {
```
[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: I had this at one point.  What is the deciding factor for you?

File: packages/neo-one-server-plugin-project/src/types.ts [hunk 2]
```diff
@@ -52,9 +65,14 @@ export const projectConfigSchema = {
     },
   },
   codegen: {
-    javascript: {
-      format: Boolean,
-      default: false,
+    language: {
+      format: typeof CodegenLanguage,
```
[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: 👍

File: packages/neo-one-server-plugin-project/src/build/generateCode.ts [hunk 3]
```diff
@@ -32,13 +32,13 @@ export const generateCode = async (
   });
 
   await fs.ensureDir(base);
-  if (project.codegen.javascript) {
-    await Promise.all([writeFile(abiPath, abi.js), writeFile(createContractPath, contract.js)]);
-  } else {
+  if (project.codegen.language === CodegenLanguage.TypeScript) {
```
[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/build/generateCode.ts]: 👍

File: packages/neo-one-server-plugin-project/src/types.ts [hunk 4]
```diff
@@ -30,13 +30,26 @@ export interface NEOTrackerRequestOptions {
...
+
 export interface ProjectConfig {
   readonly paths: {
     readonly contracts: string;
     readonly generated: string;
   };
   readonly codegen: {
-    readonly javascript: boolean;
+    readonly language: CodegenLanguage;
+    readonly uiframework: CodegenFrameworks;
```
[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: 👍

File: packages/neo-one-server-plugin-project/src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: image

[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: image

[the remark this answers is not in the document]
Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: image

File: packages/neo-one-server-plugin-project/src/types.ts [hunk 5]
```diff
@@ -52,9 +56,14 @@ export const projectConfigSchema = {
     },
   },
   codegen: {
```
Comment (dicarlo2) [inline packages/neo-one-server-plugin-project/src/types.ts]: This should be:
```
codegen: {
  language: {
    format: ['javascript', 'typescript'],
    default: 'typescript',
  },
  framework: {
    format: ['none', 'react'],
    framework: 'none',
  },
}
```

See https://github.com/mozilla/node-convict. (Find this out on your own by seeing where `projectConfigSchema` is used).

File: packages/neo-one-server-plugin-project/src/types.ts [hunk 6]
```diff
@@ -30,13 +30,17 @@ export interface NEOTrackerRequestOptions {
 
 export type RequestOptions = NetworkRequestOptions | SourceMapsRequestOptions | NEOTrackerRequestOptions;
 
+export type CodegenLanguage = 'javascript' | 'typescript';
+export type CodegenFramework = 'none' | 'react' | 'angular' | 'vue';
```
Comment (dicarlo2) [inline packages/neo-one-server-plugin-project/src/types.ts]: Remove angular and vue, they aren't possible in this PR.

Comment (davemneo) [reply packages/neo-one-server-plugin-project/src/types.ts]: 👍

**Pull request closed without merging**